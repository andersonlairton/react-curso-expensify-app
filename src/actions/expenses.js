import uuid from 'uuid';

export const addExpense = (
    {
         descricao = '',
          nota = '', 
          amount = 0, 
          createAT = 0
         } = {}) => (
    {
        type: 'ADD_EXPENSE',
        expense:
        {
            id: uuid(),
            descricao,
            nota,
            amount,
            createAT
        }
    });
//função que remove
export const removeExpense = ({ id } = {}) => (
    {
        type: 'REMOVE_EXPENSE',
        id
    }
);

//função que edita

export const editExpense = (id, updates) => (
    {
        type: 'EDIT_EXPENSE',
        id,
        updates
    }
);