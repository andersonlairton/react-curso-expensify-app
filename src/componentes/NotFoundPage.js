import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage=()=>
(
    <div>
        Error 404!
        pagina nao encontrada
        <br></br>
        <Link to="/">Voltar pra casa</Link>
        <br></br>
        <Link to="/"><img src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"/>
        </Link>
   </div>
);

export default NotFoundPage;