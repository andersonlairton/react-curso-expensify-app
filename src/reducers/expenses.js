const expenseReducerDefaultState=[];

export default (state =expenseReducerDefaultState,action)=>
{
    switch (action.type) 
    {
        case 'ADD_EXPENSE':
            return[...state,
                action.expense
            ];//retorne os dados da matriz
        case 'REMOVE_EXPENSE':
            return state.filter(({id})=> id !==action.id);//se o id for igual ao action.id resultara em falso
        case 'EDIT_EXPENSE':
            return state.map((expense)=>
            {
                if(expense.id===action.id)
                {
                    return{
                        ...expense,
                        ...action.updates
                    }
                }
                else
                {
                    return expense;
                }
            });
        default:
            return state;
    }
};

