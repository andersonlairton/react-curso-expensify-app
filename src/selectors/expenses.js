/*import moment from 'moment';
//classificando dados
export default (expenses,/*criando variaveis e filtros individuais { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createATMoment = moment(expense.createAT);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createATMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createATMoment, 'day') : true;
        // const startDateMatch= typeof startDate !=='number' || expense.createAT>=startDate;
        //const endDateMatch= typeof endDate !=='number'|| expense.createAT<=endDate;
        const textMatch = expense.descricao.toLowerCase().includes(text.toLowerCase());//pesquisando dados e convertendo eles para letras maiusculas

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) =>//classificando 
    {
        if (sortBy === 'date') {
            return a.createAT < b.createAT ? 1 : -1;
        } else if (sortBy === 'amount')//classificando as despesas por valores
        {
            return a.amount < b.amount ? 1 : -1;
        }
    });
}
*/
import moment from 'moment';

// Get visible expenses

export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
    const textMatch = expense.descricao.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1;
    }
  });
};
