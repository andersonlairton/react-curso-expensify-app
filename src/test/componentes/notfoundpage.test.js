import React from 'react';
import NotFoundPage from '../../componentes/NotFoundPage'
import {shallow} from 'enzyme';
import expenses from '../fixuters/expenses';

test('expenses notFound page',()=>
{
    const wrapper = shallow(<NotFoundPage/>);
    expect(wrapper).toMatchSnapshot();
});