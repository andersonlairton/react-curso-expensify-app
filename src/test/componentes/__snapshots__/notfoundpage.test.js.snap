// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expenses list item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Error 404! pagina nao encontrada",
        <br />,
        <Link
          replace={false}
          to="/"
>
          Voltar pra casa
</Link>,
        <br />,
        <Link
          replace={false}
          to="/"
>
          <img
                    src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
          />
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Error 404! pagina nao encontrada",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Voltar pra casa",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Voltar pra casa",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
/>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Error 404! pagina nao encontrada",
          <br />,
          <Link
            replace={false}
            to="/"
>
            Voltar pra casa
</Link>,
          <br />,
          <Link
            replace={false}
            to="/"
>
            <img
                        src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
            />
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Error 404! pagina nao encontrada",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Voltar pra casa",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Voltar pra casa",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
/>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`expenses notFound page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Error 404! pagina nao encontrada",
        <br />,
        <Link
          replace={false}
          to="/"
>
          Voltar pra casa
</Link>,
        <br />,
        <Link
          replace={false}
          to="/"
>
          <img
                    src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
          />
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Error 404! pagina nao encontrada",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Voltar pra casa",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Voltar pra casa",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
/>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Error 404! pagina nao encontrada",
          <br />,
          <Link
            replace={false}
            to="/"
>
            Voltar pra casa
</Link>,
          <br />,
          <Link
            replace={false}
            to="/"
>
            <img
                        src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
            />
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Error 404! pagina nao encontrada",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Voltar pra casa",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Voltar pra casa",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              src="https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg"
/>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "https://image.freepik.com/vetores-gratis/glitch-error-404-page-background_23-2148072533.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
